app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -748,7 +748,7 @@ public SearchTeiModel transform(TrackedEntitySearchItem searchItem, @Nullable Pr
                 }
             }
 
-            searchTei.setOnline(!searchItem.isOnline());
```

Comment:
As we don't want to do many changes in the code freeze, I think the fix might be enough, but, what do you think about adding comments to the lines in this function to explain what is doing in each step, it's a bit confusing to understand,

Thanks!

Reply:
I totally agree it was very confusing to understand, I will add some comments to this, see if It might help the next person who comes accross this code, also hopefully we will refactor/modularize  this soon  as it is within the old java search repository.